### PR TITLE
Edit events

### PR DESCRIPTION
**Pull request by jennydely**
- you can edit events
- you can edit packlists
- you can save both and it will be saved and load from local storage

#13

### CODE REVIEW

File: cypress/integration/eventplanner/editall.spec.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
+/// <reference types="cypress" />
+
+context('Editing', () => {
+  beforeEach(() => {
+    cy.visit('http://localhost:3000')
+  })
+  it('edits a packlist', () => {
```
Comment (karstenlenz) [inline cypress/integration/eventplanner/editall.spec.js:7]: This test is very long. I think you could split it into multiple separate tests. A lot of the steps seem unrelated to editing the list, you just check if some elements exist.

File: cypress/integration/eventplanner/editall.spec.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    cy.get('ul')
+      .contains('Rockharz')
+      .click({ timeout: 400 })
+      .get('img[alt="packlist"]')
+      .visit('http://localhost:3000/packlist/medieval')
+      .contains('PackList')
+      .get('button')
```
Comment (karstenlenz) [inline cypress/integration/eventplanner/editall.spec.js:14]: I think you don't necesseraly need the element type ('ul', 'button') if the text is unique. You only need to test the element type if you have some special requirements. The user doesn't care if it's a ul or some other element. Then your test would be shorter.

File: src/Packlist/PacklistForm.js [hunk 3]
```diff
@@ -1,43 +1,55 @@
...
 
-PacklistForm.propTypes = {
-  onPacklistSave: PropTypes.func.isRequired,
-}
-
-export default function PacklistForm({ onPacklistSave }) {
+export default function PacklistForm() {
   const { register, handleSubmit, reset, errors } = useForm()
-  const [items, setItems] = useState([])
+  const { editPacklistID } = useParams()
```
Comment (karstenlenz) [inline src/Packlist/PacklistForm.js:20]: The name "editPacklistID" may be misleading. It sounds like a function that edits a packlist ID. Maybe you can call it something like "currentPacklistId" or "editablePacklistId"

File: src/Packlist/PacklistForm.js [hunk 4]
```diff
@@ -1,43 +1,55 @@
...
   }
   const handlePacklistKeyDown = (event) => {
     if (event.key === 'Enter') {
       event.preventDefault()
-      addItem({ text: event.target.value, id: uuid() })
+      addItem({ item: event.target.value, itemID: uuid() })
     }
   }
+  useEffect(() => {
+    if (packlistToEdit?.packlist) setItems(packlistToEdit?.packlist)
```
Comment (karstenlenz) [inline src/Packlist/PacklistForm.js:51]: I think you don't need the "?" here. If that variable is undefined, the if statement will be false.

File: src/Packlist/PacklistForm.js [hunk 5]
```diff
@@ -143,7 +157,7 @@ export default function PacklistForm({ onPacklistSave }) {
   )
 
   function addItem(item) {
-    if (item.text?.trim().length >= 3 && item.text?.trim().length <= 20) {
+    if (item.item?.trim().length >= 3 && item.item?.trim().length <= 20) {
```
Comment (karstenlenz) [inline src/Packlist/PacklistForm.js:160]: "item.item" sounds a little bit weird. I'm not sure waht this might be. Maybe you can find some other name for this property that is more descriptive?

File: src/Packlist/PacklistPage.js [hunk 6]
```diff
@@ -62,6 +76,12 @@ export default function PackListPage() {
   }
 }
 
+const EditImg = styled.img`
+  width: 40px;
+  height: auto;
+  margin: 7px 4px;
+`
+const EditButton = styled.button``
```
Comment (karstenlenz) [inline src/Packlist/PacklistPage.js:84]: You don't need the "EditButton" component if you don't add any styles.

File: src/Packlist/usePacklistForm.js [hunk 7]
```diff
@@ -1,15 +1,20 @@
...
+  const { updatePacklist, addPacklist } = usePacklists()
 
   const showPacklistForm = () => setPacklistFormIsVisible(true)
 
   const onPacklistSave = (packlist) => {
     addPacklist(packlist)
     setPacklistFormIsVisible(false)
   }
-
+  const onPacklistSaveEdit = (packlist) => {
```
Comment (karstenlenz) [inline src/Packlist/usePacklistForm.js:14]: Is this function necessary? It looks like all it does is call updatePacklist. If that's true, you might just call updatePacklist directly.

File: src/common/GlobalStyles.js [hunk 8]
```diff
@@ -71,7 +75,26 @@ export default createGlobalStyle`
...
+  min-height: 45px;
+  justify-content: center;
+  background-image: url('data:image/svg+xml;charset=US-ASCII,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20width%3D%22292.4%22%20height%3D%22292.4%22%3E%3Cpath%20fill%3D%22%23FDE6A7%22%20d%3D%22M287%2069.4a17.6%2017.6%200%200%200-13-5.4H18.4c-5%200-9.3%201.8-12.9%205.4A17.6%2017.6%200%200%200%200%2082.2c0%205%201.8%209.3%205.4%2012.9l128%20127.9c3.6%203.6%207.8%205.4%2012.8%205.4s9.2-1.8%2012.8-5.4L287%2095c3.5-3.5%205.4-7.8%205.4-12.8%200-5-1.9-9.2-5.5-12.8z%22%2F%3E%3C%2Fsvg%3E'),
+    linear-gradient(
+      to bottom,
+      rgb(96, 99, 104) 20%,
+      rgba(96, 99, 104) 100%
+    );
+  background-repeat: no-repeat, repeat;
+  /* arrow icon position (1em from the right, 50% vertical) , then gradient position*/
```
Comment (karstenlenz) [inline src/common/GlobalStyles.js:94]: Remove comments before merging to master.

File: src/components/useEventForm.js [hunk 9]
```diff
@@ -1,16 +1,29 @@
...
     addEvent(event)
     setEventFormIsVisible(false)
   }
+  const onEventSaveEdit = (event) => {
+    const newEvent = JSON.parse(JSON.stringify(event))
+    updateEvent(newEvent)
+    setEventFormIsVisible(false)
+  }
+  const goEventBack = () =>
+    window.location.reload() && setEventFormIsVisible(false)
```
Comment (karstenlenz) [inline src/components/useEventForm.js:20]: Do you need the logical operator && ?

Would this work too?
```
const goEventBack = () =>{
    window.location.reload()
    setEventFormIsVisible(false)}
```

**Review by karstenlenz — APPROVED** [review]
This was a huge PR with many files changed. I couldn't find any real problems, just found that some parts are little bit hard to understand for me.  I left some comments, maybe you can add some variable names to make things easier to read for another developer.

File: cypress/integration/eventplanner/editall.spec.js (hunk 1 above)
Comment (jennydely) [reply cypress/integration/eventplanner/editall.spec.js:7]: I tried to make it easier, but it is difficult, to get the right element to use it.

File: cypress/integration/eventplanner/editall.spec.js (hunk 2 above)
Comment (jennydely) [reply cypress/integration/eventplanner/editall.spec.js:14]: My buttons are img so I dont have any texts on my elements..

File: src/Packlist/PacklistForm.js (hunk 3 above)
Comment (jennydely) [reply src/Packlist/PacklistForm.js:20]: I think about this, thanks

File: src/Packlist/PacklistForm.js (hunk 4 above)
Comment (jennydely) [reply src/Packlist/PacklistForm.js:51]: The useeffect should just be working, when the packlistToEdit is avaiable. I tried it without the ? and got an infinity loop :/

File: src/Packlist/PacklistForm.js (hunk 5 above)
Comment (jennydely) [reply src/Packlist/PacklistForm.js:160]: its the item of the item ;D I dont know, how to name it otherwise.. because it's not the text I need here, so I ha to use item propertie of the item, Maybe I will find another word for the item

File: src/Packlist/PacklistPage.js (hunk 6 above)
Comment (jennydely) [reply src/Packlist/PacklistPage.js:84]: I forgot to delete it, after adding the image instead of text on the button.

File: src/Packlist/usePacklistForm.js (hunk 7 above)
Comment (jennydely) [reply src/Packlist/usePacklistForm.js:14]: I will proof it, because of using hook forms, at some points I need to handle functions more complicate than normally

File: src/common/GlobalStyles.js (hunk 8 above)
Comment (jennydely) [reply src/common/GlobalStyles.js:94]: I think this comments are helpful for other people to understand, what this style is for

File: src/components/useEventForm.js (hunk 9 above)
Comment (jennydely) [reply src/components/useEventForm.js:20]: I need the logical operator, otherwise its not working.

**Comment by jennydely** [comment]
Thanks for your feedback, I will check the mentioned things

File: src/Packlist/usePacklistForm.js (hunk 7 above)
Comment (jennydely) [reply src/Packlist/usePacklistForm.js:14]: I need the return of true and it's easier to understand thats an edit function instead of just an update?
